tests/agent/actions/test_generate_data_quality_check_expectations_action.py

```diff
@@ -759,6 +759,91 @@ def mock_create_expectation(expectation, asset_id, created_via):
     assert isinstance(expectation.max_value, dict) and "$PARAMETER" in expectation.max_value
 
 
+def test_generate_completeness_forecast_expectations_action_success(
+    mock_context: CloudDataContext,
+    mocker: MockerFixture,
+    mock_metric_repository: MetricRepository,
+    mock_batch_inspector: BatchInspector,
+    mock_completeness_metrics: MockCompletenessMetrics,
+):
+    """Test that a single ExpectColumnProportionOfNonNullValuesToBeBetween expectation is created for completeness with forecast."""
+    # Setup
+    metrics = mock_completeness_metrics(30, 100)
+
+    action = GenerateDataQualityCheckExpectationsAction(
+        context=mock_context,
+        metric_repository=mock_metric_repository,
+        batch_inspector=mock_batch_inspector,
+        base_url="",
+        auth_key="",
+        organization_id=uuid.uuid4(),
+    )
+
+    # Mock the methods that would be called
+    def mock_retrieve_asset(event, asset_name):
+        return TableAsset(
+            id=TABLE_ASSET_ID,
+            name="test-data-asset",
+            table_name="test_table",
+            schema_name="test_schema",
+        )
+
+    def mock_get_metrics(data_asset):
+        return MetricRun(metrics=metrics), uuid.uuid4()
+
+    def mock_get_coverage(data_asset_id):
+        return {}
+
+    mocker.patch.object(action, "_retrieve_asset_from_asset_name", side_effect=mock_retrieve_asset)
+    mocker.patch.object(action, "_get_metrics", side_effect=mock_get_metrics)
+    mocker.patch.object(
+        action, "_get_current_anomaly_detection_coverage", side_effect=mock_get_coverage
+    )
+
+    # Mock the _create_expectation_for_asset method to capture created expectations
+    created_expectations = []
+
+    def mock_create_expectation(expectation, asset_id, created_via):
+        created_expectations.append(expectation)
```

Comment:
total nit: probably cleaner to just assert what a spy was called with rather than implement a mechanism to capture these